### PR TITLE
Add ability to mute a test fixture and mute krb5

### PR DESCRIPTION
**Pull request by alpar-t**
Relates to #40624 
There was no clear way to mute the fixture itself, so this PR also has a change to allow doing that besides muting the flaky fixture.

### CODE REVIEW

File: buildSrc/src/main/java/org/elasticsearch/gradle/testfixtures/TestFixturesPlugin.java [hunk 1]
```diff
@@ -128,6 +135,16 @@ public void apply(Project project) {
 
     }
 
+    private void conditionTaskByType(TaskContainer tasks, TestFixtureExtension extension, Class<? extends DefaultTask> taskClass) {
```
Comment (mark-vieira) [inline buildSrc/src/main/java/org/elasticsearch/gradle/testfixtures/TestFixturesPlugin.java]: Can we give this a bit more of a descriptive name? Especially all these nested lambdas are a bit hard to follow.

Perhaps `skipTasksWithDisabledFixtures`.

File: buildSrc/src/main/java/org/elasticsearch/gradle/testfixtures/TestFixturesPlugin.java [hunk 2]
```diff
@@ -94,21 +96,26 @@ public void apply(Project project) {
...
+        conditionTaskByType(tasks, extension, getTaskClass("org.elasticsearch.gradle.test.RestIntegTestTask"));
+        conditionTaskByType(tasks, extension, TestingConventionsTasks.class);
+        conditionTaskByType(tasks, extension, ComposeUp.class);
+
         if (dockerComposeSupported(project) == false) {
             project.getLogger().warn(
                 "Tests for {} require docker-compose at /usr/local/bin/docker-compose or /usr/bin/docker-compose " +
                     "but none could be found so these will be skipped", project.getPath()
             );
-            disableTaskByType(tasks, getTaskClass("com.carrotsearch.gradle.junit4.RandomizedTestingTask"));
```
Comment (mark-vieira) [inline buildSrc/src/main/java/org/elasticsearch/gradle/testfixtures/TestFixturesPlugin.java]: So does this mean we will no longer disable tests when Docker isn't available?

**Review by mark-vieira — COMMENTED** [review]
One question and suggest but otherwise LGTM. 👍

File: buildSrc/src/main/java/org/elasticsearch/gradle/testfixtures/TestFixturesPlugin.java (hunk 2 above)
Comment (atorok) [reply buildSrc/src/main/java/org/elasticsearch/gradle/testfixtures/TestFixturesPlugin.java]: We will do  it via  `onlyIf` since `buidFixture` will be auto disabled when there's no docker.

Comment (mark-vieira) [reply buildSrc/src/main/java/org/elasticsearch/gradle/testfixtures/TestFixturesPlugin.java]: And by "auto-disabled" we simply won't add the requisite task dependencies (the early `return`)?

Maybe it'd be clearer to explicitly disable `buildFixture` in the same way if docker isn't available?

Comment (atorok) [reply buildSrc/src/main/java/org/elasticsearch/gradle/testfixtures/TestFixturesPlugin.java]: We do disable `buildDioxture` as in `setEnabled(false)` when docker is not available. 
We do add the task dependencies, it seems sightly less confusing that way since the dependencies will always be the same. We do need to create the task so the configuration phase still runs correctly even without docker

Comment (mark-vieira) [reply buildSrc/src/main/java/org/elasticsearch/gradle/testfixtures/TestFixturesPlugin.java]: I think I'm missing something. The way it's wired up now, if `dockerComposeSupported() == false` then we simply short-circuit the plugin, which results in the test task dependencies never being setup but at no point is the `buildFixture` task explicitly disabled. We just never run the tests, because we never add the dependencies.
